### PR TITLE
Camera class

### PR DESCRIPTION
**Pull request by bseto**
Created the videoStream Class that encapsulates the camera thread.
Integrated the class into Main.cpp and edited the Makefile to accommodate for it

### CODE REVIEW

**Comment by bseto** [comment]
@enochtsang

File: SDL/include/threadClass.hpp [hunk 1]
```diff
@@ -1,4 +1,7 @@
+#ifndef THREADCLASS_H
+#define THREADCLASS_H
 //#include <pthread.h>
+//
```
Comment (ExtraGravity) [inline SDL/include/threadClass.hpp]: Delete this line?

File: SDL/include/videoStream.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        VideoStream();
+        void signalToQuit();
+        IplImage *getFrame();
+        bool imageReady();
+
+
+    protected:
+        void InternalThreadEntry();
+
+    private:
```
Comment (ExtraGravity) [inline SDL/include/videoStream.hpp:22]: private members should start with m_

File: SDL/src/Main.cpp [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
 //for the rasperry pi
 #ifndef INT64_C
 #define INT64_C(c) (c ## LL)
 #define UINT64_C(c) (c ## ULL)
 #endif
-
 extern "C" {
     #include <SDL.h>
-    #include <SDL_thread.h>
+//    #include <SDL_thread.h>
```
Comment (ExtraGravity) [inline SDL/src/Main.cpp]: remove this line? Don't think we plan to use it in the future.

File: SDL/src/videoStream.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+VideoStream::VideoStream()
+{
+    bufferNumber = 1; 
+    updatedImage = false;
+
+}
+
+void VideoStream::signalToQuit()
+{
+    quit = 1;
```
Comment (ExtraGravity) [inline SDL/src/videoStream.cpp]: `quit = true;` might be more clear?

File: SDL/include/videoStream.hpp [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#ifndef VIDEOSTREAM_H
+#define VIDEOSTREAM_H 
+#include "threadClass.hpp"
```
Comment (ExtraGravity) [inline SDL/include/videoStream.hpp:4]: Nitpicky, but I think a empty line between 
`#define VIDEOSTREAM_H`
and 
`#include "threadClass.hpp"` 
would look nicer.

File: SDL/include/videoStream.hpp [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+
+    protected:
+        void InternalThreadEntry();
+
+    private:
+        Mat frame;
+        bool updatedImage;
+        int bufferNumber;
+        IplImage threadImage1;
```
Comment (ExtraGravity) [inline SDL/include/videoStream.hpp]: This is triple buffer right?

File: SDL/src/Main.cpp [hunk 7]
```diff
@@ -1,22 +1,20 @@
 
 #include <cv.h>
 #include "opencv2/opencv.hpp"
-#include <SongLoader.h>
-#include <SongPlayer.h>
-
+#include "SongLoader.h"
```
Comment (ExtraGravity) [inline SDL/src/Main.cpp:4]: Extra line here as well would look nicer I think.

**Comment by ExtraGravity** [comment]
Nicely done, so much cleaner. So beautiful :')

File: SDL/include/videoStream.hpp (hunk 6 above)
Comment (bseto) [inline SDL/include/videoStream.hpp]: Yep, we have 3 of these threadImages for the triple buffer.

File: SDL/src/Main.cpp (hunk 7 above)
Comment (bseto) [inline SDL/src/Main.cpp:4]: between SongLoader.h and SongPlayer.h? Sorry its hard to tell from this comment^

Comment (ExtraGravity) [inline SDL/src/Main.cpp:4]: NEVER MIND, read it wrong. Disregard my comment.

**Comment by gordon-cheung** [comment]
SDL mixer needs to be removed from makefile before it compiles, otherwise the files compile fine and  BackupCamera works (Camera and musicplayer functioning as should)

**Pull request merged**